core/dataset_operations.cpp

```diff
@@ -25,30 +55,37 @@ DataArray histogram(const DataConstProxy &sparse,
   if (binEdges.dtype() != dtype<double> ||
       sparse.coords()[dim].dtype() != DType::Double)
     throw std::logic_error("Histogram is only available for double type.");
-  auto coord = sparse.coords()[dim];
-  auto edgesSpan = binEdges.values<double>();
-  if (!std::is_sorted(edgesSpan.begin(), edgesSpan.end()))
-    throw std::logic_error("Bin edges should be sorted to make the histogram.");
-  auto resDims{sparse.dims()};
-  auto len = binEdges.dims()[dim] - 1;
-  resDims.resize(resDims.index(dim), len);
-  Variable result = makeVariableWithVariances<double>(resDims, units::counts);
-  for (scipp::index i = 0; i < sparse.dims().volume(); ++i) {
-    const auto &coord_i = coord.sparseValues<double>()[i];
-    auto curRes = result.values<double>().begin() + i * len;
-    for (const auto &c : coord_i) {
-      auto it = std::upper_bound(edgesSpan.begin(), edgesSpan.end(), c);
-      if (it != edgesSpan.end() && it != edgesSpan.begin())
-        ++(*(curRes + (--it - edgesSpan.begin())));
-    }
-  }
-  std::copy(result.values<double>().begin(), result.values<double>().end(),
-            result.variances<double>().begin());
 
-  std::map<Dim, Variable> coords;
-  coords.emplace(dim, binEdges);
-  return {std::move(result), std::move(coords),
-          std::map<std::string, Variable>()};
+  auto result = apply_and_drop_dim(
+      sparse,
+      [](const DataConstProxy &_sparse, const Dim _dim,
```

Comment:
I think we should avoid names starting with `_`, there are some rules about when they are reserved (which I think you are not strictly violating, but why get close).

Reply:
Do you have a better suggestion? Ideally I'd want to avoid using a completely different name since they are actually the same value.

Reply:
Put the `_` at the end of the variable name.